app/models/declaration.rb

```diff
@@ -79,4 +81,25 @@ def ineligible_for_funding_reason
       state_reason
     end
   end
+
+  def duplicate_declarations
+    self
+      .class
+      .joins(application: %i[user course])
+      .where(user: { trn: application.user.trn })
+      .where.not(user: { trn: nil })
+      .where.not(id:)
```

Comment:
could we also check that the user isn't the same as per ECF:
```
.where.not(user_id:, id:)
```

Reply:
I think this:
```
.where.not(user: { trn: nil, id: application.user_id })
```
would only check `user_id` if `trn` is nil and should be two separate where statements:
```
.where.not(user: { trn: nil })
.where.not(user: { id: application.user_id })
```
Or I'm messing up the logic in my head.

Reply:
discussed in person: let's separate just for readability and remove includes